src/Features/CSharp/Portable/ConvertLinq/ConvertForEachToLinqQuery/ToCountConterter.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.ConvertLinq.ConvertForEachToLinqQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertLinq.ConvertForEachToLinqQuery
+{
+    internal sealed class ToCountConterter : AbstractToMethodConverter
+    {
+        public ToCountConterter(
+            ForEachInfo<ForEachStatementSyntax, StatementSyntax> forEachInfo,
+            ExpressionSyntax selectExpression,
+            ExpressionSyntax modifyingExpression,
+            SyntaxTrivia[] trivia)
+            : base(forEachInfo, selectExpression, modifyingExpression, trivia) { }
+
+        protected override string MethodName => nameof(Enumerable.Count);
+
+        // Checks that the expression is "0".
+        protected override bool CanReplaceInitialization(ExpressionSyntax expression, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => expression is LiteralExpressionSyntax literalExpression && literalExpression.Token.ValueText == "0";
```

Comment:
just text checking? not http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/SemanticModel.cs,736 ?

Reply:
It can do the same but it seems that the current approach is simpler.

---
In reply to: [195849961](https://github.com/dotnet/roslyn/pull/27130#discussion_r195849961) [](ancestors = 195849961)